### PR TITLE
build: add TRACY_SUPPORT cmake option and fix vcpkg baseline

### PR DESCRIPTION
**Pull request by FlayaN**
Had to rename the cmake option to `TRACY_SUPPORT` due to `TRACY_ENABLE` being hardcoded to defined when using cmake target. Added a undef to that if `TRACY_SUPPORT` is set to `OFF` (which is now default)

### CODE REVIEW

File: src/State.h [hunk 1]
```diff
@@ -144,8 +145,10 @@ class State
 
 	inline ~State()
 	{
+#ifdef TRACY_ENABLE
```
Comment (Pentalimbed) [inline src/State.h:148]: When no TRACY_ENABLE, `TracyD3D11Destroy` is interpreted as empty strings, so no need

Comment (Pentalimbed) [reply src/State.h:148]: Matter of fact I don't think tracy is designed to require manual TRACY_ENABLE checks anywhere

Comment (FlayaN) [reply src/State.h:148]: Caused a warning without this
image

File: include/PCH.h [hunk 2]
```diff
@@ -31,6 +31,10 @@ void* operator new[](size_t size, size_t alignment, size_t alignmentOffset, cons
 #define WIN32_LEAN_AND_MEAN
 #include <Windows.h>
 
+#ifndef TRACY_SUPPORT
+	#undef TRACY_ENABLE
```
Comment (Pentalimbed) [inline include/PCH.h:35]: Why undef? Where is the def?

Comment (FlayaN) [reply include/PCH.h:35]: Hardcoded here :'(
image

File: src/State.h (hunk 1 above)
Comment (Pentalimbed) [reply src/State.h:148]: uhhhhh I think we should turn this warning off

**Pull request merged**